### PR TITLE
Replace gsl-lite with span-lite

### PR DESCRIPTION
**Pull request by jcelerier**
For #28

### CODE REVIEW

File: include/quickcpplib/span.hpp [hunk 1]
```diff
@@ -65,13 +65,13 @@ QUICKCPPLIB_NAMESPACE_END
 
 #define gsl_CONFIG_DEFAULTS_VERSION 1
```
Comment (BurningEnlightenment) [inline include/quickcpplib/span.hpp:67]:
```suggestion
```
I guess this is obsolete now, right?

File: .gitmodules [hunk 2]
```diff
@@ -26,3 +26,6 @@
 	path = include/quickcpplib/byte
 	url = https://github.com/martinmoene/byte-lite.git
 	branch = master
+[submodule "include/quickcpplib/span-lite"]
+	path = include/quickcpplib/span-lite
+	url = https://github.com/martinmoene/span-lite
```
Comment (BurningEnlightenment) [inline .gitmodules:31]:
```suggestion
```
This seems to be redundant with the other span-lite submodule.

**Review by BurningEnlightenment — CHANGES_REQUESTED** [review]
I found some minor nits, but we'll need to wait for Niall's approval.

File: include/quickcpplib/span.hpp (hunk 1 above)
Comment (ned14) [reply include/quickcpplib/span.hpp:67]: Looks likely.

**Comment by ned14** [comment]
I pretty much agree with @BurningEnlightenment's feedback. Thanks for the PR!

File: .gitmodules (hunk 2 above)
Comment (jcelerier) [reply .gitmodules]: done, a small git mishap !

File: include/quickcpplib/span.hpp (hunk 1 above)
Comment (jcelerier) [reply include/quickcpplib/span.hpp]: done

**Comment by ned14** [comment]
@BurningEnlightenment Will this need some changes to the vcpkg port?

**Comment by BurningEnlightenment** [comment]
Yeah, I'll take care of that. Will probably take a few days due to the review process, etc.

**Comment by jcelerier** [comment]
thanks ! :)

**Comment by ned14** [comment]
https://github.com/ned14/outcome/runs/7141879761?check_suite_focus=true suggests this PR broke Outcome CI?

**Comment by BurningEnlightenment** [comment]
I will take a look in an hour or so.

**Comment by BurningEnlightenment** [comment]
> https://github.com/ned14/outcome/runs/7141879761?check_suite_focus=true suggests this PR broke Outcome CI?

This looks like the an issue I encountered while updating the vcpkg port  (_not_ #32). I pushed  9cdcd45d9036247dcade3d3ab1ff4adbe14b1927 onto the master branch to fix it. So the CI should be green again next time.

**Pull request merged**